File: POProjekt3/rosliny/Guarana.py
from Roslina import Roslina

class Guarana(Roslina):
    
    def __init__(self):
        pass

    def __init__(self, swiat, poz, wiek, inicj, moc):

        super(Roslina, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeWiek(wiek)
        super().changeInicjatywa(inicj)

        s = "G"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    def __init__(self, swiat, poz):

        super(Roslina, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeInicjatywa(0)
        super().zmienSile(0)
        super().changeWiek(0)

        s = "G"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    #--------------------------------------------

    def rysowanie(self):
        print('G')

    def getZnak(self):
        return 'G'
    
    def dodawanaSilaPoZjedzeniu(self):
        return 3

File: VirtualWorld-Cpp/Virtual-World/Guarana.cpp
#include "Owca.h"
#include "Zwierze.h"
#include <iostream>
#include "Organizm.h"
#include "Trawa.h"
#include "Roslina.h"
#include "Guarana.h"

Guarana::Guarana()
{
}

Guarana::Guarana(Swiat* swiat, std::vector<unsigned int> poz, int wiek, int inicj, int moc)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeWiek(wiek);
	changeInicjatywa(inicj);

	getSwiat()->edytujPlansze(poz[0], poz[1], GUARANA);
}

Guarana::Guarana(Swiat* swiat, std::vector<unsigned int> poz)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeInicjatywa(0);
	zmienSile(0);
	changeWiek(0);

	getSwiat()->edytujPlansze(poz[0], poz[1], GUARANA);
}

void Guarana::rysowanie()
{
	std::cout << GUARANA;
}

char Guarana::getZnak()
{
	return GUARANA;
}

int Guarana::dodawanaSilaPoZjedzeniu()
{
	return 3;
}

Guarana::~Guarana()
{
}